src/MongoDB/ReadPreference.c

```diff
@@ -202,17 +202,105 @@ static PHP_METHOD(ReadPreference, getTagSets)
 	}
 } /* }}} */
 
+static HashTable* php_phongo_readpreference_get_properties_hash(zval* object, bool is_debug TSRMLS_DC) /* {{{ */
+{
+	php_phongo_readpreference_t* intern;
+	HashTable*                   props;
+	const char*                  modeString = NULL;
+	const bson_t*                tags;
+	mongoc_read_mode_t           mode;
+
+	intern = Z_READPREFERENCE_OBJ_P(object);
+
+	PHONGO_GET_PROPERTY_HASH_INIT_PROPS(is_debug, intern, props, 3);
+
+	if (!intern->read_preference) {
+		return props;
+	}
+
+	tags = mongoc_read_prefs_get_tags(intern->read_preference);
+	mode = mongoc_read_prefs_get_mode(intern->read_preference);
+
+	switch (mode) {
+		case MONGOC_READ_PRIMARY:
+			modeString = "primary";
+			break;
+		case MONGOC_READ_PRIMARY_PREFERRED:
+			modeString = "primaryPreferred";
+			break;
+		case MONGOC_READ_SECONDARY:
+			modeString = "secondary";
+			break;
+		case MONGOC_READ_SECONDARY_PREFERRED:
+			modeString = "secondaryPreferred";
+			break;
+		case MONGOC_READ_NEAREST:
+			modeString = "nearest";
+			break;
+		default: /* Do nothing */
+			break;
+	}
+
+	if (modeString) {
+#if PHP_VERSION_ID >= 70000
+		zval z_mode;
+
+		ZVAL_STRING(&z_mode, modeString);
+		zend_hash_str_update(props, "mode", sizeof("mode") - 1, &z_mode);
+#else
+		zval* z_mode;
+
+		MAKE_STD_ZVAL(z_mode);
+		ZVAL_STRING(z_mode, modeString, 1);
+		zend_hash_update(props, "mode", sizeof("mode"), &z_mode, sizeof(z_mode), NULL);
+#endif
+	}
+
+	if (!bson_empty0(tags)) {
+		/* Use PHONGO_TYPEMAP_NATIVE_ARRAY for the root type since tags is an
+		 * array; however, inner documents and arrays can use the default. */
+		php_phongo_bson_state state = PHONGO_BSON_STATE_INITIALIZER;
+		state.map.root_type         = PHONGO_TYPEMAP_NATIVE_ARRAY;
+
+		php_phongo_bson_to_zval_ex(bson_get_data(tags), tags->len, &state);
+#if PHP_VERSION_ID >= 70000
+		Z_ADDREF(state.zchild);
+		zend_hash_str_update(props, "tags", sizeof("tags") - 1, &state.zchild);
+#else
+		Z_ADDREF_P(state.zchild);
+		zend_hash_update(props, "tags", sizeof("tags"), &state.zchild, sizeof(state.zchild), NULL);
+#endif
+		zval_ptr_dtor(&state.zchild);
+	}
+
+	if (mongoc_read_prefs_get_max_staleness_seconds(intern->read_preference) != MONGOC_NO_MAX_STALENESS) {
+		long maxStalenessSeconds = mongoc_read_prefs_get_max_staleness_seconds(intern->read_preference);
+#if PHP_VERSION_ID >= 70000
+		zval z_max_ss;
```

Comment:
Is this declaration kosher, since the previous line calls `mongoc_read_prefs_get_max_staleness_seconds()`?

I compiled locally with dev flags and didn't notice any warnings.

Reply:
Sure, this is fine. You can declare a variable, and assign a default value, even if that comes from a function call.

Reply:
Perhaps my previous issues with a declaration after a statement was do to using macros that look like function calls but actually do other things :)